.github/workflows/buildIOS.yml

```diff
@@ -0,0 +1,155 @@
+        run: echo "IS_PODFILE_SAME_AS_MANIFEST=${{ hashFiles('ios/Podfile.lock') == hashFiles('ios/Pods/Manifest.lock') }}" >> "$GITHUB_OUTPUT"
+
+      - name: Install cocoapods
+        uses: nick-fields/retry@3f757583fb1b1f940bc8ef4bf4734c8dc02a5847
+        if: steps.pods-cache.outputs.cache-hit != 'true' || steps.compare-podfile-and-manifest.outputs.IS_PODFILE_SAME_AS_MANIFEST != 'true' || steps.setup-node.outputs.cache-hit != 'true'
+        with:
+          timeout_minutes: 10
+          max_attempts: 5
+          command: scripts/pod-install.sh
+
+      - name: Decrypt provisioning profiles
+        run: |
+          cd ios
+          provisioningProfile=''
+          if [ '${{ inputs.type }}' == 'release' ]; then
+            provisioningProfile='NewApp_AppStore'
+          else
+            provisioningProfile='NewApp_AdHoc'
+          fi
+          echo "Using provisioning profile: $provisioningProfile"
+          gpg --quiet --batch --yes --decrypt --passphrase="$LARGE_SECRET_PASSPHRASE" --output "$provisioningProfile.mobileprovision" "$provisioningProfile.mobileprovision.gpg"
+          gpg --quiet --batch --yes --decrypt --passphrase="$LARGE_SECRET_PASSPHRASE" --output "${provisioningProfile}_Notification_Service.mobileprovision" "${provisioningProfile}_Notification_Service.mobileprovision.gpg"
+        env:
+          LARGE_SECRET_PASSPHRASE: ${{ secrets.LARGE_SECRET_PASSPHRASE }}
+
+      - name: Decrypt code signing certificate
+        run: cd ios && gpg --quiet --batch --yes --decrypt --passphrase="$LARGE_SECRET_PASSPHRASE" --output Certificates.p12 Certificates.p12.gpg
+        env:
+          LARGE_SECRET_PASSPHRASE: ${{ secrets.LARGE_SECRET_PASSPHRASE }}
+
+      - name: Build iOS ${{ inputs.type }} app
+        id: build
+        run: |
+          lane=''
+          if [ '${{ inputs.type }}' == 'release' ]; then
+            lane='build'
+          else
+            lane='build_adhoc'
+          fi
+
+          bundle exec fastlane ios "$lane"
+
+          # Reload environment variables from GITHUB_ENV
+          # shellcheck disable=SC1090
+          source "$GITHUB_ENV"
+
+          {
+            # ipaPath and dsymPath are environment variables set within the Fastfile
+            echo "IPA_PATH=$ipaPath"
+            echo "IPA_FILE_NAME=$(basename "$ipaPath")"
+            echo "DSYM_PATH=$dsymPath"
+            echo "DSYM_FILE_NAME=$(basename "$dsymPath")"
+          } >> "$GITHUB_OUTPUT"
+
+      - name: Upload iOS build artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: ios-artifact-ipa
+          path: ${{ steps.build.outputs.IPA_PATH }}
+
+      - name: Upload iOS debug symbols artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: ios-artifact-dsym
+          path: ${{ steps.build.outputs.DSYM_PATH }}
+
+      - name: Upload iOS sourcemaps
+        uses: actions/upload-artifact@v4
+        with:
+          name: ios-sourcemaps-artifact
```

Comment:
Should it be `ios-artifact-sourcemaps`?

Reply:
Also do we do anything with this?

Reply:
I kept it as `ios-sourcemaps-artifact` intentionally, because that's what it was called before. I'm not sure but I think we might download this artifact as part of some debug/development tools. I can look into it

Reply:
In the only other reference in this repo, it's just uploaded to the GitHub Release. I think it might be downloaded from there manually to debug issues reported in #newdot-quality though

Reply:
Ah - The only reason I said is because I'm not sure we will download it `pattern: ios-artifact-*`? But if it's downloaded elsewhere already, that sounds good

Reply:
ended up renaming this to `ios-artifact-sourcemaps` for consistency. It will still get uploaded to the GitHub Release